Create ui-c8y-1020-0-10-smart-rule-contains-unexpected-child-mo-6409-graft-release-cd-6458.md (#2130)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1020-0-10-smart-rule-contains-unexpected-child-mo-6409-graft-release-cd-6458.md

---
date: ""

> title: Smart Rule contains unexpected child MO (#6409) [GRAFT][release/cd] (#6458)

Author: The Claude AI suggested to use the following title:
```suggestion
title: Smart rules can contain unexpected child managed objects
```

Reviewer: I think that title does not need to be changed.

> title: Smart Rule contains unexpected child MO (#6409) [GRAFT][release/cd] (#6458)

Reviewer: ```suggestion
title: Smart Rule list contains children column when there is child assets or child devices assign to MO
```

> title: Smart rules list now contains a children column if child assets are assigned to a managed object

Reviewer 2: ```suggestion
title: Smart rules list now always shows the children column properly if child assets are assigned to a managed object
```

> title: Smart Rule list contains children column when there is child assets or child devices assign to MO

Reviewer 2: ```suggestion
title: Smart rules list now contains a children column if child assets are assigned to a managed object
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YdSEScrEC
    label: Cockpit
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-57897
version: 1020.0.10
---

> Smart Rule contains unexpected child MO (#6409) [GRAFT][release/cd] (#6458)

Author: The Claude AI suggested to use the following body:
```suggestion
In some cases, smart rules could incorrectly contain child managed objects that were not expected. This caused the smart rules to behave in unintended ways and could lead to incorrect results. The issue has been resolved and smart rules now only contain the expected child managed objects. Existing smart rules that contained unexpected child managed objects will need to be reviewed and potentially updated to ensure they work as intended.
```

Reviewer: I would suggest a little bit different description.

> Smart Rule contains unexpected child MO (#6409) [GRAFT][release/cd] (#6458)

Reviewer: ```suggestion
When smart rule is created in context of parent device, during smart rule creation it is possible to select child device. When smart rule is saved it automatically becomes active for selected asset. Then on list of smart rules it is possible to see that it is active for child device.
```

Reviewer: @Airdroplet what do you think? Is it ok?

> When smart rule is created in context of parent device, during smart rule creation it is possible to select child device. When smart rule is saved it automatically becomes active for selected asset. Then on list of smart rules it is possible to see children column and that it is active for child device.

Reviewer 2: ```suggestion
If a smart rule is created for a parent device, during smart rule creation you can select for which child devices the smart rule should become active. When the smart rule is saved it automatically becomes active for the selected assets. In the smart rules list,  a children column is displayed showing for which child device the smart rule is active.
```

> If a smart rule is created for a parent device, during smart rule creation you can select for which child devices the smart rule should become active. When the smart rule is saved it automatically becomes active for the selected assets. In the smart rules list,  a children column is displayed showing for which child device the smart rule is active.

Reviewer 2: @pmich-sag As this is a fix, what was the bug :)? Sounds more like a feature.

Reviewer: Description of the [issue](https://cumulocity.atlassian.net/browse/MTM-57897) is a little bit different than what was to be fix. You can see comments there, and it turned out, that parent-device has assigned some childAsset, and children column was visible. But it was not visible for child-device which was assigned to parent-device as childDevice. Now implementation checks both of these properties.
![MTM-57897_children_column](https://github.com/SoftwareAG/c8y-docs/assets/133014746/439d8faa-5060-4452-861d-220222404f25)

> If a smart rule is created for a parent device, during smart rule creation you can select for which child devices the smart rule should become active. When the smart rule is saved it automatically becomes active for the selected assets. In the smart rules list,  a children column is displayed showing for which child device the smart rule is active.

Reviewer 2: ```suggestion
In the smart rules list of a group or a device, the children column, which shows for which child device the smart rule is active, did not always show up properly. This issue has been fixed and the children column is now always displayed if the smart rule is activated for a child device.
```
